src/pages/RFOX/hooks/useAffiliateRevenueQuery.ts

```diff
@@ -0,0 +1,62 @@
+import axios from 'axios'
+import { getConfig } from 'config'
+import { useMemo } from 'react'
+import { useQuery } from 'wagmi/query'
+
+type AffiliateRevenueQueryKey = [
+  'affiliateRevenue',
+  { startTimestamp: bigint; endTimestamp: bigint },
+]
+
+type UseAffiliateRevenueQueryProps = {
+  startTimestamp: bigint
+  endTimestamp: bigint
+}
+
+export const getAffiliateRevenueQueryKey = (
+  startTimestamp: bigint,
+  endTimestamp: bigint,
+): AffiliateRevenueQueryKey => ['affiliateRevenue', { startTimestamp, endTimestamp }]
+
+export const affiliateRevenueQueryFn = async ({
+  queryKey,
+}: {
+  queryKey: AffiliateRevenueQueryKey
+}) => {
+  const [, { startTimestamp, endTimestamp }] = queryKey
```

Comment:
suggestion (non-blocking): this should not take a queryKey arg, but be a good ol' async fn taking its own args in.

rationale: this isn't type-safe (can't narrow args with `skipToken` if we ever have optionals),`queryKey` and `queryFn` args may be different, and semantically speaking a key should only be used for key-ing purposes.

nitpick: in addition to the suggestion above, rename this function to match its new "good ol' async function" nature  e.g `fetchAffiliateRevenue`

A few other occurences of one of/both points above in https://github.com/shapeshift/web/pull/7162 and across this PR as well, but happy to tackle as cleanup as part of https://github.com/shapeshift/web/issues/7171

Reply:
actioned, glad to get a solid pattern locked in now so this doesnt spread